compiler_opt/rl/corpus.py

```diff
@@ -122,6 +122,10 @@ def filter(self, p: re.Pattern):
     """Filters module specs, keeping those which match the provided pattern."""
     self._module_specs = [ms for ms in self._module_specs if p.match(ms.name)]
 
+  @property
+  def modules(self):
+    return list(self._module_specs)
```

Comment:
why constrain as list?

Reply:
you mean the list()? So a copy is returned, to prevent mutation of the internal list

Reply:
a, ok. so `get_modules_copy`? and it's not a @property then?